pkg/operator/configobservation/etcd/observe_etcd.go

```diff
@@ -197,11 +199,16 @@ func isPendingReady(bucket string, podName string, scalingName string, podLister
 		klog.Errorf("isPendingReady: error getting pod %#v", err)
 		return false
 	}
+	if len(pod.Spec.InitContainers) < numberOfInitContainers {
+		return true
+	}
 	if len(pod.Status.InitContainerStatuses) < 2 {
 		klog.Infof("isPendingReady: waiting for init cert containers to pass")
 		return false
 	}
-	if pod.Status.InitContainerStatuses[1].State.Terminated != nil && pod.Status.InitContainerStatuses[1].State.Terminated.ExitCode == 0 && pod.Status.InitContainerStatuses[2].State.Running != nil {
+	certsInitContainerNumber := getInitContainerNumber(pod, "certs")
+	membershipInitContainerNumber := getInitContainerNumber(pod, "membership")
+	if pod.Status.InitContainerStatuses[certsInitContainerNumber].State.Terminated != nil && pod.Status.InitContainerStatuses[certsInitContainerNumber].State.Terminated.ExitCode == 0 && pod.Status.InitContainerStatuses[membershipInitContainerNumber].State.Running != nil {
```

Comment:
If `getInitContainerNumber` returned -1, it will panic. Perhaps you should check for non-negative values here?

Reply:
https://github.com/openshift/cluster-etcd-operator/pull/51/commits/5ba7b80a7ca2a7eed73cc062cd0f93926ca69f2c#diff-30a0cbc1713c8b08f59b22c1302264cfR202

This check will allow for stopping the panic?

Reply:
Yeah, you are pretty sure if there 3 containers, one of them is going to be certs and other is membership. Then, that's fine.

Reply:
Clearly this code is better than using hardcoded [1] and [2] as was done previously. I am marking this as resolved.

Reply:
yes, this will allow us to add `wait-for-kube` initcontainer without breaking the operator. Thanks!